lib/rubocop/runner.rb

```diff
@@ -98,8 +98,11 @@ def process_file(file)
       file_started(file)
 
       offenses = file_offenses(file)
-      formatter_set.file_finished(file, offenses)
-      offenses
+      output_offenses = offenses.select do |offense|
+        !@options[:fail_level_output] || considered_failure?(offense)
```

Comment:
Perhaps you could avoid looping through the offenses here by pulling the condition out of the `select`? It might even better express its intent. For example like this:

```rb
unless @options[:fail_level_output]
  offenses = offenses.select { |offense| considered_failure?(offense) }
end
```